File: ReformaAgraria/Models/Event.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ReformaAgraria.Models
{
    public class Event: BaseEntity<int>
    {
        public Event() { }
        
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public override int Id { get; set; }        

        public string Description { get; set; }                

        public DateTime StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        public string FkRegionId { get; set; }

        public string ResultDescription { get; set; }

        public string Attendees { get; set; }

        [ForeignKey("FkRegionId")]
        public Region Region { get; set; }

        public string FkEventTypeId { get; set; }

        [ForeignKey("FkEventTypeId")]
        public EventType EventType { get; set; }
    }
}

File: ReformaAgraria/ClientApp/app/helpers/event.ts
import { Event } from '../models/gen/event';
import { CalendarEvent, CalendarEventAction } from 'angular-calendar';
import { EventColor } from 'calendar-utils';

import * as moment from 'moment';

export class EventHelper {

    static serialize(calEvent: CalendarEvent<Event>): Event {
        let event: Event = {
            id: calEvent.meta.id,
            description: calEvent.meta.description,
            startDate: calEvent.start,
            endDate: calEvent.end,
            fkRegionId: calEvent.meta.fkRegionId,
            fkEventTypeId: calEvent.meta.fkEventTypeId
        };
        return event;
    }

    static deserialize(event: Event, actions?: CalendarEventAction[]): CalendarEvent<Event> {
        let evColor = this.getEventColor(event);
        let color = {
            primary: evColor,
            secondary: evColor
        };

        event.startDate = event.startDate ? moment.utc(event.startDate).local().toDate() : null;
        event.endDate = event.endDate ? moment.utc(event.endDate).local().toDate() : null;

        let calEvent: CalendarEvent<Event> = {
            start: event.startDate,
            end: event.endDate,
            title: event.eventType.name,
            color: color,
            draggable: true,
            resizable: {
                beforeStart: true,
                afterEnd: true
            },
            actions: actions ? actions : [],
            meta: event
        };
        return calEvent;
    }

    static serializeMany(calEvents: CalendarEvent<Event>[]): Event[] {
        let events = [];
        calEvents.forEach(calEvent => {
            events.push(this.serialize(calEvent));
        });
        return events;
    }

    static deserializeMany(events: Event[], actions?: CalendarEventAction[]): CalendarEvent[] {
        let calEvents = [];
        events.forEach(event => {
            calEvents.push(this.deserialize(event, actions));
        });
        return calEvents;
    }

    static getEventColor(event: Event): string {
        switch (event.region.type) {
            case 2:
                return '#09833A'
            case 3:
                return '#BBD86E'
            case 4:
                return '#C3976A'
            default:
                return '#C3976A'
        }

    }
}